features/vfork.c

```diff
@@ -0,0 +1,45 @@
+// This is for the Meson config stage to check if vfork() acts like
+// ksh expects.  We need vfork() to pause the parent and share memory
+// with the child until the child execs or exits.
+//
+// These systems do pause the parent and share memory:
+// - FreeBSD: https://www.freebsd.org/cgi/man.cgi?vfork
+// - Illumos: https://illumos.org/man/vfork
+// - Linux: http://man7.org/linux/man-pages/man2/vfork.2.html
+// - NetBSD: http://netbsd.gw.com/cgi-bin/man-cgi?vfork
+//
+// This system doesn't share memory:
+// - OpenBSD: https://man.openbsd.org/vfork
+
+#include <errno.h>
+#include <sys/wait.h>
+#include <unistd.h>
+
```

Comment:
This test is different from the [legacy feature test](https://github.com/att/ast/blob/e770c77e9816e156c6df4a455e71b5f9fff79310/src/lib/libast/features/lib#L242-L269) for vfork(). I presume this means legacy test was not testing desired functionality on OpenBSD.

Reply:
Ok, I see [similar legacy feature test](https://github.com/att/ast/blob/e770c77e9816e156c6df4a455e71b5f9fff79310/src/lib/libast/features/lib#L271-L285) that defines `_real_vfork` macro.

Reply:
I didn't know about the old feature test. It looks simpler than mine, and might have worked.

I didn't want to bring back the old code that had too many macros (at least _lib_vfork, _real_vfork, _hdr_vfork, _sys_vfork, _lib_fork). Instead, I made _lib_vfork like _lib_posix_spawn. There's only _lib_posix_spawn = 1 if we use it, or _lib_posix_spawn = 0 if we don't.

Reply:
I think it makes sense to restore [old feature test](https://github.com/att/ast/blob/e770c77e9816e156c6df4a455e71b5f9fff79310/src/lib/libast/features/lib#L271-L285) than writing a new one.